open_event/static/js/admin/event/scheduling.js

```diff
@@ -0,0 +1,749 @@
+            } else {
+                addSessionToUnscheduled($sessionElement);
+            }
+
+        },
+        ondropdeactivate: function (event) {
+            var $trackDropZone = $(event.target);
+            var $sessionElement = $(event.relatedTarget);
+            $trackDropZone.removeClass("drop-now").removeClass("drop-active");
+            if (!$sessionElement.hasClass("scheduled")) {
+                $sessionElement.css({
+                    "-webkit-transform": "",
+                    "transform": "",
+                    "background-color": ""
+                }).removeData("x").removeData("y").tooltip("hide").attr("data-original-title", "");
+            }
+        }
+    });
+}
+
+/**
+ * This callback called after sessions and tracks are processed.
+ * @callback postProcessCallback
+ */
+/**
+ * Process the tracks and sessions data loaded from the server into in-memory data stores
+ * @param {object} tracks The tracks json object
+ * @param {object} sessions The sessions json object
+ * @param {postProcessCallback} callback The post-process callback
+ */
+function processTrackSession(tracks, sessions, callback) {
+
+    var topTime = moment.utc({hour: time.start.hours, minute: time.start.minutes});
+
+    _.each(sessions, function (session) {
+
+        var startTime = moment.utc(session.start_time);
+        var endTime = moment.utc(session.end_time);
+        var duration = moment.duration(endTime.diff(startTime));
+
+        var top = minutesToPixels(moment.duration(moment.utc({
+            hour: startTime.hours(),
+            minute: startTime.minutes()
+        }).diff(topTime)).asMinutes(), true);
+
+        var dayString = startTime.format("Do MMMM YYYY"); // formatted as eg. 2nd May 2013
+
+        if (!_.includes(days, dayString)) {
+            days.push(dayString);
+        }
+
+        /**
+         * @type {{id: number, title: string, top: number, duration: number, track_id: number|null, start_time: Moment, end_time: Moment}}
+         */
+        var sessionObject = {
+            id: session.id,
+            title: session.title,
+            top: top,
+            duration: Math.abs(duration.asMinutes()),
+            track_id: session.track.id,
+            start_time: startTime,
+            end_time: endTime
+        };
+
+        var dayIndex = _.indexOf(days, dayString);
+
+        if (_.isArray(sessionsStore[dayIndex])) {
+            sessionsStore[dayIndex].push(sessionObject);
+        } else {
+            sessionsStore[dayIndex] = [sessionObject]
+        }
+    });
+
+    _.each(tracks, function (track) {
+        var tracksObject = {
+            name: track.name,
+            id: track.id
+        };
+        if (!_.includes(tracksStore, tracksObject)) {
+            tracksStore.push(tracksObject);
+        }
+    });
+
+    loadDateButtons();
+    callback();
+}
+
+/**
+ * Load the date selection button onto the DOM
+ */
+function loadDateButtons() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['loadDateButtons' is defined but never used.](https://www.codacy.com/app/triplez/open-event-orga-server/file/2668227635/issues/source?bid=3233864&fileBranchId=3324174#l623)